prog/vm/github_runner.rb

```diff
@@ -285,6 +285,27 @@ def setup_info
 
     if vm
       vm.private_subnets.each { _1.incr_destroy }
+
+      # If the runner is not assigned any job and we destroy it after a
+      # timeline, the workflow_job is nil, in that case, we want to be able to
+      # see journalctl output to debug if there was any problem with the runner
+      # script.
+      #
+      # We also want to see the journalctl output if the runner script failed.
+      #
+      # Hence, the condition is added to check if the workflow_job is nil or
+      # the conclusion is failure.
+      if (job = github_runner.workflow_job).nil? || job.fetch("conclusion", "failure") == "failure"
```

Comment:
Recently, we discovered that the job would hang when the GitHub lost connection with the runner script. In such cases, customers might cancel operations. Therefore, these runs will be labeled as 'cancelled' instead of 'failure'. Let's save all logs except those marked 'success'.

```suggestion
      if (job = github_runner.workflow_job).nil? || job["conclusion"] != "success"
```